src/Lykke.Service.Stellar.Api.Services/Horizon/HorizonService.cs

```diff
@@ -135,94 +145,117 @@ public async Task<bool> AccountExists(string address)
         {
-            if (txEnvelope?.Tx?.Operations == null || txEnvelope.Tx.Operations.Length < 1 ||
-                txEnvelope.Tx.Operations[0].Body == null)
+            if (txEnvelope.Discriminant.InnerValue == EnvelopeType.EnvelopeTypeEnum.ENVELOPE_TYPE_TX_V0)
+            {
+                if (txEnvelope?.V0?.Tx.Operations == null || txEnvelope?.V0?.Tx.Operations.Length < 1 ||
+                    txEnvelope?.V0?.Tx.Operations[0].Body == null)
+                {
+                    throw new HorizonApiException($"Failed to extract first operation from transaction.");
+                }
+
+                var operation = txEnvelope?.V0?.Tx.Operations[0].Body;
+                return operation;
+            }
+
+            if (txEnvelope.Discriminant.InnerValue == EnvelopeType.EnvelopeTypeEnum.ENVELOPE_TYPE_TX)
             {
-                throw new HorizonApiException($"Failed to extract first operation from transaction.");
+                if (txEnvelope?.V1?.Tx.Operations == null || txEnvelope?.V1?.Tx.Operations.Length < 1 ||
+                    txEnvelope?.V1?.Tx.Operations[0].Body == null)
+                {
+                    throw new HorizonApiException($"Failed to extract first operation from transaction.");
+                }
+
+                var operation = txEnvelope?.V1?.Tx.Operations[0].Body;
+                return operation;
             }
 
-            var operation = txEnvelope.Tx.Operations[0].Body;
-            return operation;
+
+            throw new HorizonApiException($"Failed to extract first operation from transaction.");
         }
 
-        public string GetMemo(TransactionDetails tx)
+        public string GetMemo(TransactionResponse tx)
         {
             if ((StellarSdkConstants.MemoTextTypeName.Equals(tx.MemoType, StringComparison.OrdinalIgnoreCase) ||
                 StellarSdkConstants.MemoIdTypeName.Equals(tx.MemoType, StringComparison.OrdinalIgnoreCase)) &&
-                !string.IsNullOrEmpty(tx.Memo))
+                !string.IsNullOrEmpty(tx.Memo.ToXdr().Text))
             {
-                return tx.Memo;
+                return tx.Memo.ToXdr().Text;
             }
 
             return null;
         }
 
-        public string GetTransactionHash(StellarBase.Generated.Transaction tx)
+        public string GetTransactionHash(stellar_dotnet_sdk.xdr.Transaction tx)
         {
-            var writer = new ByteWriter();
-
-            // Hashed NetworkID
-            writer.Write(Network.CurrentNetworkId);
-
-            // Envelope Type - 4 bytes
-            EnvelopeType.Encode(writer, EnvelopeType.Create(EnvelopeType.EnvelopeTypeEnum.ENVELOPE_TYPE_TX));
-
-            // Transaction XDR bytes
-            var txWriter = new ByteWriter();
-            StellarBase.Generated.Transaction.Encode(txWriter, tx);
-            writer.Write(txWriter.ToArray());
+            throw new NotImplementedException();
```

Comment:
Is this method needed if it's not implemented?

Reply:
fixed